### ISSUE TITLE
std and var reductions mismatch with torch by a scale factor

### ISSUE THREAD

**Issue by nsorabaTT** [tenstorrent/tt-metal#19478]
**Describe the bug**
std and var reductions for the same tensor do not match that from torch. They are off by a constant factor (sqrt(2) and 2 respectively).

**To Reproduce**
```python
import random
import torch
import ttnn

# Parameters
random.seed(0)
rank, tensor_shape, dim, keepdim = 2, [2, 2], 0, True

# Setup
device = ttnn.open_device(device_id=0)

# Run
torch_tensor = torch.tensor([3.0, 5.0, 40, 9.0]).reshape(tensor_shape)
ttnn_tensor = ttnn.from_torch(torch_tensor, layout=ttnn.TILE_LAYOUT, device=device)
ttnn_result_device = ttnn.var(ttnn_tensor, dim=dim, keepdim=keepdim)
ttnn_result_host = ttnn.from_device(ttnn_result_device)
ttnn_result = ttnn.to_torch(ttnn_result_host)
torch_result = torch.var(torch_tensor, dim=dim, keepdim=keepdim)

# Verify
assert torch.allclose(ttnn_result, torch_result, atol=0.2, rtol=0.05), "ttnn: {} vs torch: {}".format(ttnn_result, torch_result)
print(torch_tensor)

# Cleanup
ttnn.close_device(device)
``` 

**Expected behavior**
`all_close` matches and no assertion is raised.

**Please complete the following environment information:**
- OS: Ubuntu 20.04
- Commit: `ddb21729db23d6eecddac29db5802b6c2509c35f`

**Comment by nsorabaTT** [comment]
This issue is triaged to incorrect padding for std and var.

Currently, std and var operations are padded just like sum or mean, i.e. with zeros. When the mean of the tensor is non-zero to start with, padding zeros shifts the mean and hence impacts the std and var calculations.


Two approaches possible to address this:
1. Calculate the mean and pad with that when using `fill_implicit_tile_padding`.
2. Make a modification to the reader kernel to read the mean instead of padded values when feeding the circular buffers.

(1) adds a new op overhead but can be optimized as mean is anyway calculated as part of var. (1) has a limitation of not supporting std/var over a given dimension. This is because padding will need to be different for different rows, something `fill_implicit_tile_padding` does not support.
(2) Will work for all cases. It does not add any "op" overhead. However, there will be overhead in the reader kernel for inputs which required padding.

A third option is to have LLK mask some of the values in the src tile. However, this feature doesn't to exist*.

*To be confirmed.

**Comment by cmaryanTT** [comment]
While looking at this, please also validate that we can match torch in general for bf16, and measure the error for fp32.

**Comment by bbradelTT** [comment]
@fplavecTT what's the status of this? Did your recent std/var updates fix this? Is all that's left to measure against torch and report the results here?

**Comment by fplavecTT** [comment]
@bbradelTT Yes, this is resolved, just need to measure/report results.

**Comment by fplavecTT** [comment]
The mismatch by a scale factor was fixed with PR #41838 in both the new Welford implementation and in the legacy implementation. Unit tests confirmed good PCC and other error metrics for bf16.

@cmaryanTT For requested fp32 accuracy, we get the results below with Welford (which is now the default path).

For small tensors, like the ones in `test_reduction.py` we get the following errors:
```
ttnn.std — float32 worst-case across all dims/shapes:

  ┌───────────┬────────────┐
  │  metric   │ worst case │
  ├───────────┼────────────┤
  │ max_abs   │ 1.0005e-03 │
  ├───────────┼────────────┤
  │ max_rel   │ 9.2163e-04 │
  ├───────────┼────────────┤
  │ frobenius │ 5.5453e-04 │
  ├───────────┼────────────┤
  │ max_ulp   │ 11,434     │
  └───────────┴────────────┘

  ttnn.var — float32 worst-case:

  ┌───────────┬────────────┐
  │  metric   │ worst case │
  ├───────────┼────────────┤
  │ max_abs   │ 2.6103e-03 │
  ├───────────┼────────────┤
  │ max_rel   │ 1.8581e-03 │
  ├───────────┼────────────┤
  │ frobenius │ 1.1068e-03 │
  ├───────────┼────────────┤
  │ max_ulp   │ 22,893     │
  └───────────┴────────────┘
```

For a larger tensor that is not tile aligned `(3, 6, 40, 63, 20)` we get the following errors:

```
ttnn.std (3, 6, 40, 63, 20) cases:

  ┌──────────┬──────────┬──────────┬───────────┬─────────┐
  │   dim    │ max_abs  │ max_rel  │ frobenius │ max_ulp │
  ├──────────┼──────────┼──────────┼───────────┼─────────┤
  │ None     │ 4.82e-04 │ 4.82e-04 │ 4.82e-04  │ 8083    │
  ├──────────┼──────────┼──────────┼───────────┼─────────┤
  │ -1       │ 1.39e-03 │ 1.12e-03 │ 5.46e-04  │ 13690   │
  ├──────────┼──────────┼──────────┼───────────┼─────────┤
  │ -2       │ 6.37e-04 │ 5.65e-04 │ 3.55e-04  │ 8508    │
  ├──────────┼──────────┼──────────┼───────────┼─────────┤
  │ (-2, -1) │ 8.43e-04 │ 8.39e-04 │ 5.45e-04  │ 9969    │
  └──────────┴──────────┴──────────┴───────────┴─────────┘

  ttnn.var (3, 6, 40, 63, 20) cases:

  ┌──────────┬──────────┬──────────┬───────────┬─────────┐
  │   dim    │ max_abs  │ max_rel  │ frobenius │ max_ulp │
  ├──────────┼──────────┼──────────┼───────────┼─────────┤
  │ None     │ 9.63e-04 │ 9.63e-04 │ 9.63e-04  │ 16163   │
  ├──────────┼──────────┼──────────┼───────────┼─────────┤
  │ -1       │ 4.07e-03 │ 2.23e-03 │ 1.09e-03  │ 27357   │
  ├──────────┼──────────┼──────────┼───────────┼─────────┤
  │ -2       │ 1.45e-03 │ 1.13e-03 │ 7.11e-04  │ 16815   │
  ├──────────┼──────────┼──────────┼───────────┼─────────┤
  │ (-2, -1) │ 1.73e-03 │ 1.68e-03 │ 1.09e-03  │ 19805   │
  └──────────┴──────────┴──────────┴───────────┴─────────┘
```

These all seem like expected floating-point rounding errors.

AI-generated test script attached

[test_std_var_fp32_errors.py](https://github.com/user-attachments/files/27407709/test_std_var_fp32_errors.py)

.

**Comment by fplavecTT** [comment]
Just a note that PR #45319 significantly improves FP32 accuracy over the results above.
